### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Lidya1234**
-Added Game and player files
-Added Game, validator, and player classes.
-Made the project to work like a tictac toe game

### CODE REVIEW

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

#### Hey @Lidya1234,  your project is looking really good, however, to accomplish as the actual project, you're required to do these changes, listed below:
 
 
## [REQUIRED CHANGES]
- [x] No puts/print and get.chomp outside `main.rb` file. ([screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from Pathwright)

Behaviour, like rules and moves: Couldn't be reviewed due⤵
- [x] Your game isn't working, please fix it;
image

## Summary
✨ Nice classes usage 💯
✨ Well done creating single responsibility methods ⚙
✨ Descriptive PR ✅

## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Lidya Ghebreigzabher Okubamichael** 
Nice job the implementing the required changes :muscle:  :muscle: 
We still need to fix some issues before you can go ahead with your next challenge:

- [x] You should remove the .gitkeep file, it's not necessary anymore.
- [x] Game must show the player name when you request an action, you might replace "Enter your choice" with something like "<player_name>, enter your choice".
- [x] Game must show the player's name when you inform that the action was invalid
- [x] Game must show the player's name when the player has won
- [x] Make sure the player's name has at least one character.
- [x] When the game was a draw, replace "Game Over" with "Game was a draw" or similar

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+      @validate.move(player_turn)
+    end
+    choice
+  end
+
+  def movement(position, sign, turn)
+    player_turn = turn.odd? ? @player1 : @player2
+    loop do
+      if @board[position - 1] == ' '
```
Comment (kevcap) [inline lib/game.rb:68]: Here at line 69 you can add `system ""cls"` to improve UX of your game;

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

#### Hey @Lidya1234,  your project is looking really good, however, to accomplish as the actual project, you're required to do these changes, listed below:
 
 
## [REQUIRED CHANGES]

- [x] As @andresporras3423 said: When the game was a draw, replace "Game Over" with "Game was a draw" or similar.
image


## [OPTIONAL CHANGES] 
By opening your game.rb, and adding system "cls", you clear the terminal everytime you game is drawn, so It gets prettier
image

## Summary
✨ Nice classes usage 💯
✨ Well done creating single responsibility methods ⚙
✨ Descriptive PR ✅

## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Review by SpaYco — APPROVED** [review]
# Project Approved

thank you for making the required changes, your project is now approved!

### please don't forget to merge, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**